atomate/vasp/workflows/tests/test_bulk_modulus_workflow.py

```diff
@@ -0,0 +1,248 @@
+
+struct_si = PymatgenTest.get_structure("Si")
+ndeformations = 6
+deformations = [(np.identity(3)*(1+x)).tolist() for x in np.linspace(-0.05, 0.05, ndeformations)]
+
+
+class TestBulkModulusWorkflow(unittest.TestCase):
+    """This test will either actually run VASP (if VASP_CMD is set) or artificially pass on outputs
+    (if not VASP_CMD) and test the whole bulk modulus workflow and its implementation and outputs
+    for an example calculation for silicon.
+
+    note for the developer of the test:
+    This tests can be run in two modes if not VASP_CMD:
+    1. first all inputs and outputs of all deformations are present in which case
+        in each folder a trimmed version of task_doc.json will be generated so that
+    2. once task_doc.json is present, VaspRun can be skipped where task_doc.json is
+        available and their "inputs" and "outputs" folders can be removed
+    """
+    @classmethod
+    def setUpClass(cls):
+        if not SETTINGS.get("PMG_VASP_PSP_DIR"):
+            SETTINGS["PMG_VASP_PSP_DIR"] = os.path.join(module_dir, "..", "..", "tests", "reference_files")
+            print('This system is not set up to run VASP jobs. '
+                  'Please set PMG_VASP_PSP_DIR variable in your ~/.pmgrc.yaml file.')
+
+        cls.scratch_dir = os.path.join(module_dir, "scratch")
+        cls.wf_config = {"deformations": deformations,
+                            "vasp_cmd": ">>vasp_cmd<<", "db_file": ">>db_file<<"}
+
+        cls.wf = wf_bulk_modulus(struct_si, cls.wf_config)
+
+
+
+
+    def setUp(self):
+        if os.path.exists(self.scratch_dir):
+            shutil.rmtree(self.scratch_dir)
+        os.makedirs(self.scratch_dir)
+        os.chdir(self.scratch_dir)
+        try:
+            self.lp = LaunchPad.from_file(os.path.join(db_dir, "my_launchpad.yaml"))
+            self.lp.reset("", require_password=False)
+        except:
+            raise unittest.SkipTest(
+                'Cannot connect to MongoDB! Is the database server running? '
+                'Are the credentials correct?')
+
+
+
+    def tearDown(self):
+        if not DEBUG_MODE:
+            shutil.rmtree(self.scratch_dir)
+            self.lp.reset("", require_password=False)
+            db = self._get_task_database()
+            for coll in db.collection_names():
+                if coll != "system.indexes":
+                    db[coll].drop()
+
+
+
+    def _simulate_vasprun(self, wf):
+        no_vasp_ref_dirs = {}
+        fake_vasp_ref_dirs = {}
+        for i in range(2, ndeformations+2):
+            if os.path.exists(os.path.join(reference_dir, str(i), "inputs")):
+                fake_vasp_ref_dirs["bulk_modulus deformation {}".format(i-2)] = os.path.join(reference_dir, str(i))
+            else:
+                no_vasp_ref_dirs["bulk_modulus deformation {}".format(i-2)] = os.path.join(reference_dir, str(i))
+
+        fake_vasp_ref_dirs["structure optimization"] =  os.path.join(reference_dir, "1")
+
+        print("here")
```

Comment:
Quick comment: I strongly discourage the removal of prints, but rather use logging.debug to log debug messages. These are helpful when you need to find out what went wrong and you can just do a run with a logging.basicConfig(level=logging.DEBUG). 

That said, of course something like print("here") is completely useless from the debugging standpoint. The messages need to be a lot more useful.